### PR TITLE
[2020-u1-rls] Finiteness checker fix

### PR DESCRIPTION
**Pull request by Alexsandruss**
# Description
Changes proposed in this pull request:

- Change of OR operator from bitwise to logical in all data values finiteness check
- Addition of lost memory allocation error check in summing

### CODE REVIEW

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/ea57dd7a-4236-f1ba-a40a-8cdcd4b723a1

File: service/kernel/data_management/finiteness_checker.cpp [hunk 1]
```diff
@@ -129,6 +140,7 @@ DataType computeSumAVX512Impl(size_t nDataPtrs, size_t nElementsPerPtr, const Da
 
     daal::services::internal::TArray<DataType, avx512> partialSumsArr(nTotalBlocks);
     DataType * pSums = partialSumsArr.get();
+    if (!pSums) return getInf<DataType>();
```
Comment (ShvetsKS) [inline service/kernel/data_management/finiteness_checker.cpp]: It`s not clear. Why we return inf in case of memory allocation error.

Comment (Alexsandruss) [reply service/kernel/data_management/finiteness_checker.cpp]: If infinity is returned as sum, finiteness checker goes to next step (finiteness check of all elements). Comment added.

Comment (ShvetsKS) [reply service/kernel/data_management/finiteness_checker.cpp]: seems status should be propagated here (by parameter for example )

**Pull request merged**